lrrbot/commands/static.py

```diff
@@ -1,86 +1,143 @@
+import itertools
 import logging
 import aiomas
 import random
 import re
 import irc.client
+import sqlalchemy
 
 import lrrbot.decorators
 from common.config import config
-from lrrbot import storage
+import common.utils
 from lrrbot.command_parser import Blueprint
 
 blueprint = Blueprint()
 log = logging.getLogger(__name__)
 
-def generate_docstring():
-	inverse_responses = {}
-	for command, data in storage.data["responses"].items():
-		response = data["response"]
-		if isinstance(response, (tuple, list)):
-			response = tuple(response)
-		inverse_responses.setdefault((response, data["access"]), [])
-		inverse_responses[(response, data["access"])] += [command]
+ADVICE = 1
+
+ACCESS_ANY = 0
+ACCESS_SUB = 1
+ACCESS_MOD = 2
+
+def _get_command_id(bot, conn, command):
+	if isinstance(command, int):
+		return command
+	else:
+		aliases = bot.metadata.tables["commands_aliases"]
+		query = sqlalchemy.select(aliases.c.command_id).where(aliases.c.alias == command)
+		row = conn.execute(query).first()
+		if row is None:
+			return None
+		else:
+			return row[0]
+
+def get_response(bot, command):
+	responses = bot.metadata.tables["commands_responses"]
+
+	with bot.engine.connect() as conn:
+		command_id = _get_command_id(bot, conn, command)
+		if command_id is None:
+			return None
+
+		query = sqlalchemy.select(responses.c.response).where(responses.c.command_id == command_id)
+		row = common.utils.pick_random_elements(conn.execute(query), 1)[0]
+
+		if row is None:
+			return None
+		else:
+			return row[0]
```

Comment:
We can use [`Result.scalars()`](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.scalars) to remove a layer of nesting.
```suggestion
		return common.utils.pick_random_elements(conn.execute(query).scalars(), 1)[0]
```